### DISCUSSION TITLE
Vite Support

### DISCUSSION THREAD

**Discussion by nelsieborja** [nrwl/nx#5497] [Feature Requests]
Is it possible to include Vite support and use it with React for example?

**Comment by nemonemi** [comment]
I find the current webpack solution sluggish with slow start-up times.
It would be nice to have the option to use some of these faster solutions that are available at the moment.

**Comment by quirin-buechner-mdctec** [comment]
would be awesome if we could build nx apps with vite

**Comment by jaysoo** [comment]
We are considering it but no roadmap. The current focus is webpack 5 support, which is already in  Nx 12.6 (https://nx.dev/latest/react/guides/webpack-5).

We are cleaning up and optimizing the webpack 5 setup. After that we can look into adding an option for vite.

**Reply by BhaskaranR** [reply]
any updates on official vitejs support?

**Reply by nxnzesrc** [reply]
Bump

**Comment by sledorze** [comment]
ViteJs development is so fast compared to Webpack, also eager to have it supported officially

**Comment by JakeGinnivan** [comment]
BTW, I started a Vite plugin while working at WANews. https://www.npmjs.com/package/@wanews/nx-vite

**Comment by sledorze** [comment]
@JakeGinnivan started to use it :)
Have you been able to make Vite work with Storybook in your Nx Repo?

**Reply by JakeGinnivan** [reply]
Yep, using https://github.com/eirslett/storybook-builder-vite. Am also using pnpm so needed to use the workarounds in https://github.com/eirslett/storybook-builder-vite/issues/55

**Reply by sledorze** [reply]
I have not been able yet to make it work so far (after a few hours) but more importantly failed to have a working build ( same issue as https://github.com/vitejs/vite/issues/2139 )

**Reply by chrisbarnes** [reply]
@JakeGinnivan would you be able to share the relevant details of your Vite/Storybook setup? I'd love to get this up and running in our Nx repo.

**Reply by alexandrzavalii** [reply]
@JakeGinnivan would be interested to see the code for the setup as well.

**Reply by JakeGinnivan** [reply]
This is our main.js

```
const rootMain = require('../../../.storybook/main')
const { dirname } = require('path')

module.exports = {
    ...rootMain,

    core: { ...rootMain.core, builder: 'storybook-builder-vite' },
    stories: [
        ...rootMain.stories,
        '../src/**/*.stories.mdx',
        '../src/**/*.stories.@(js|jsx|ts|tsx)',
    ],
    addons: [...rootMain.addons],
    async viteFinal(config) {
        // customize the Vite config here

        // https://github.com/eirslett/storybook-builder-vite/issues/55
        config.root = dirname(require.resolve('storybook-builder-vite'))
        config.server.fsServe = undefined

        config.optimizeDeps.include = [
            ...(config.optimizeDeps?.include ?? []),
            'react-router-dom'
        ]

        return config
    },
}
```

We have recently added Chakra UI though and it's broken our setup, I haven't had a chance to look into it.

I also am looking at https://www.ladle.dev/blog/introducing-ladle/ because it aligns with Vite better and won't have the same dependency issues.

**Reply by alexandrzavalii** [reply]
@JakeGinnivan How do you make aliases work in vite build?

We have them defined in `tsconfig.json`

**Reply by james-elicx** [reply]
> @JakeGinnivan How do you make aliases work in vite build?
> 
> We have them defined in `tsconfig.json`

@alexandrzavalii you need to add the aliases to your vite config too. The following is pulled from one of my projects:

```js
export default defineConfig(() => ({
  ...
  resolve: {
    alias: {
      '@/': join(PACKAGE_ROOT, 'src') + '/',
      '@assets/': join(PACKAGE_ROOT, 'assets') + '/',
    },
  },
  ...
}));
```

**Comment by AbdallahAbis** [comment]
any updates on this @jaysoo ?

**Comment by dsmileybrainly** [comment]
Any update on this? Our devs are getting more and more excited about Vite (and Snowpack).

**Comment by MurrayJack** [comment]
Joining the update party

**Comment by mlp73** [comment]
Also interested in some updates :)

**Comment by parzival418** [comment]
Also interested!

**Comment by AbbasSrour** [comment]
Add me to the gang, Vite support would be super awesome

**Comment by patricknazar** [comment]
I'd like to see this too.

**Comment by nonameolsson** [comment]
Seems like Storybook is switching focus from Webpack to Vite 👍🏻 
https://storybook.js.org/blog/first-class-vite-support-in-storybook/

I tried the upgrade but didn't manage to get it to work (yet).

**Comment by yukimotochern** [comment]
Vite would be great~~~~~

**Comment by a777med** [comment]
Would be great.

**Comment by zanettin** [comment]
as @vsavkin mentioned on the roadmap 16 (https://github.com/nrwl/nx/discussions/12836) outlook: it is considered 🎉

**Comment by Boniqx** [comment]
Needing this too.

**Comment by simonnepomuk** [comment]
[Docs](https://nx.dev/packages/vite) for the official plugin already exist, but it's currently not working see #13197

**Comment by jaysoo** [comment]
The Vite plugin has been released in December 2022: https://nx.dev/packages/vite.

You can use it with standalone React apps: `npx create-nx-workspace@latest --preset=react-standalone`.

Just posting the update here in case anyone lands here from a Google search.

**Reply by skyisuniverse** [reply]
what about vite + vitest support for next.js package in nx? currently only jest is supported

**Reply by stewones** [reply]
looking for a vite alternative for next too =/

**Comment by josipfrljic** [comment]
How to include assets in app from lib, when use @nrwl/vite? With webpack it works with setup in workspace.json (assets:[glob,ect], but in case of vite it doesn't work. Anyone have an idea?

**Comment by terwer** [comment]
is there a way to build cli application using @nrwl/vite? I mean setup a bin entry , the default is a lib
